### ISSUE TITLE
Smart Moving/RenderPlayerAPI Support (Fix suggested in post)

### ISSUE THREAD

**Issue by Razorskills** [aidancbrady/Mekanism#1237]
Hey Aidan, I was wondering if you could add support for smart moving render?
(When I say that I don't mean the smart moving mod, just RenderPlayerAPI)

This is what Divisor said about supporting it:

--------------------------

The solution to this problem is that every mod that uses Minecraft Forge's"Item.getArmorModel" hook returns a "api.player.model.ModelPlayer" with Smart Moving friendly coding instead of a"net.minecraft.client.model.ModelBiped" in case Render Player API is present.

--------------------------

Machine Muse was able to get it working with modular powersuits quite well. here is the ticket that we made to get it fixed:

https://github.com/MachineMuse/MachineMusePowersuits/issues/463

I think There is info in that post about how smart moving needs to be hooked to the player, might be useful or not..

Here is some source code links that might help:
https://github.com/MachineMuse/MachineMusePowersuits/blob/master/src/minecraft/net/machinemuse/powersuits/client/render/item/SMovingArmorModel.scala

https://github.com/MachineMuse/MachineMusePowersuits/blob/master/src/minecraft/api/player/model/ModelPlayer.java

Here is a link to RenderPlayerAPI: http://www.minecraftforum.net/topic/1261354-172api-render-player-api/

If you could fix this, I would be so happy! But If not il try living with these picture:
https://www.dropbox.com/sh/4gmiw5trvtei5gt/Zc9SHduW3X

Basically the jetpack (or any model added to the player) will face north always unless hooked into RenderPlayerAPI if it is present.

**Comment by crunkazcanbe** [comment]
I would Love to see Nelson work with smart moving! The jetpack looks bad when smart moving is installed

**Comment by aidancbrady** [comment]
@Razorskills, could I have an update on this?  I'd love to add Smart Moving support, but I wouldn't be surprised if RenderPlayerAPI changed again since 1.6.4.

**Comment by Razorskills** [comment]
@aidancbrady There is some discussion here on the topic: https://github.com/Tonius/SimplyJetpacks/issues/84

I will have to look into it some more though

**Comment by crunkazcanbe** [comment]
This would be so nice. There are a few mods supporting smart moving , and if this supported it it would mean mod pack devs would be that much closer to being able to include smart moving in their packs .

**Comment by aidancbrady** [comment]
I think I'm going to have to close this - it seems like too much effort for compatible with only one mod.  I will happily accept a PR though that includes a fix.
